infra/core/host/appservice.bicep

```diff
@@ -87,6 +88,41 @@ resource appService 'Microsoft.Web/sites@2022-03-01' = {
       configAppSettings
     ]
   }
+
+  resource configAuth 'config' = if (!(empty(appSettings.AZURE_CLIENT_APP_ID))) {
+    name: 'authsettingsV2'
+    properties: {
+      globalValidation: {
+        requireAuthentication: true
+        unauthenticatedClientAction: 'RedirectToLoginPage'
+        redirectToProvider: 'azureactivedirectory'
+      }
+      identityProviders: {
+        azureActiveDirectory: {
+          enabled: true
+          registration: {
+            clientId: appSettings.AZURE_CLIENT_APP_ID
+            clientSecretSettingName: !empty(appSettings.AZURE_CLIENT_APP_SECRET) ? 'AZURE_CLIENT_APP_SECRET' : ''
+            openIdIssuer: appSettings.AZURE_AUTHENTICATION_ISSUER_URI
+          }
+          login: {
+            loginParameters: ['scope=openid profile email offline_access api://${appSettings.AZURE_SERVER_APP_ID}/access_as_user']
```

Comment:
I'd like to see this as a parameter to the appservice.bicep file so users could override if they need to.

Reply:
We need these permissions at minimum, so it might be an "and" rather than an "or"